gpu_bdb/bdb_tools/q28_utils.py

```diff
@@ -60,18 +70,33 @@ def read_tables(config, c=None):
     return pr_df
 
 
-def gpu_hashing_vectorizer(x):
-    vec = HashingVectorizer(n_features=N_FEATURES,
-                            alternate_sign=alternate_sign,
-                            ngram_range=ngram_range,
-                            norm=norm,
-                            preprocessor=preprocessor
-     )
+def hashing_vectorizer(x):
+
+    if isinstance(x, cudf.Series):
+        vectorizer = HashingVectorizer
+        preprocessor = lambda s:s.str.lower()
+    else:
+        vectorizer = SKHashingVectorizer
+        preprocessor = lambda s:s.lower()
+
+    vec = vectorizer(
+        n_features=N_FEATURES,
+        alternate_sign=alternate_sign,
+        ngram_range=ngram_range,
+        norm=norm,
+        preprocessor=preprocessor
+    )
+
     return vec.fit_transform(x)
 
 
 def map_labels(ser):
-    output_ser = cudf.Series(cudf.core.column.full(size=len(ser), fill_value=2, dtype=np.int32))
+
+    if isinstance(ser, cudf.Series):
+        output_ser = cudf.Series(cudf.core.column.full(size=len(ser), fill_value=2, dtype=np.int32))
+    else:
+        output_ser = pd.Series(2, index=ser.index, dtype=np.int32)
```

Comment:
We should have similar behavior here.  We should not use cudf specific API when not needed. 

Lets use cupy  and numpy arrray here.e 


```python
pd.Series(np.full(shape=len(ser), fill_value=2, dtype=np.int32))
```

```python
cudf.Series(cp.full(shape=len(ser), fill_value=2,  dtype=cp.int32))))
```